src/Workflow/MoveJatsAssets.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentStore\Workflow;
+
+use FluentDOM\DOM\Element;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Psr7\UriNormalizer;
+use GuzzleHttp\Psr7\UriResolver;
+use InvalidArgumentException;
+use League\Flysystem\AdapterInterface;
+use League\Flysystem\FilesystemInterface;
+use Libero\ContentApiBundle\Model\PutTask;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\Workflow\Event\Event;
+use UnexpectedValueException;
+use function count;
+use function explode;
+use function GuzzleHttp\Promise\all;
+use function GuzzleHttp\Psr7\uri_for;
+use function implode;
+use function in_array;
+use function is_callable;
+use function is_resource;
+use function Libero\ContentApiBundle\stream_hash;
+use function preg_match;
+use function sprintf;
+
+final class MoveJatsAssets implements EventSubscriberInterface
+{
+    private const HAS_MIMETYPE_ATTRIBUTE = [
+        'graphic',
+        'inline-graphic',
+        'inline-media',
+        'inline-supplementary-material',
+        'media',
+        'supplementary-material',
+    ];
+
+    private $client;
+    private $filesystem;
+    private $origin;
+
+    public function __construct(string $origin, FilesystemInterface $filesystem, ClientInterface $client)
+    {
+        $this->origin = $origin;
+        $this->filesystem = $filesystem;
+        $this->client = $client;
+
+        if (!is_callable([$filesystem, 'getPublicUrl'])) {
```

Comment:
any interface that can be checked rather than a method existence? There may not be one.

Reply:
No, it's a bit nasty... 😐

Reply:
Turns out that doesn't work as it _is_ callable, just throws an exception. Changed to 90e62ab7bb53c5fed5f28832a6bb99a1f68adc80, which works but is even worse.

Reply:
Not sure of a good solution to this. Maybe should revert https://github.com/libero/content-store/pull/9#discussion_r282095484...

Reply:
Done in b85996c.